Pocket/pocket-ios
# Add Analytics tracking to v7 -> v8 migration

## Summary
Add new tracking types and fire them from PocketKits and SaveToPocketKit

## References 
https://getpocket.atlassian.net/browse/IN-1277

## Implementation Details
Add new Event type, System. Other event types relate to some manner of user interaction and were therefore unsuitable for use here in a purely passive event.
A System Event is qualified by it's SystemEventType (Which as of right now only supports a UserMigration event)
To support the UserMigration event a UserMigrationState enum was also added to add granularity to the stage of migration the user is currently at.

## Test Steps

- Log into a v7 instance of Pocket.
- Close the app and update to v8
- Launch the app or use the SaveToPocketKit extension
- Check for suitable Analytics events.

## PR Checklist:
